airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoDbInitialLoadRecordIterator.java

```diff
@@ -128,7 +141,7 @@ private Bson buildFilter() {
             case OBJECT_ID -> new BsonObjectId(new ObjectId(state.id()));
             case INT -> new BsonInt32(Integer.parseInt(state.id()));
             case LONG -> new BsonInt64(Long.parseLong(state.id()));
-            case BINARY -> parseBinaryIdString(state.id());
+            case BINARY -> parseBinaryIdString(state.id(), state.binarySubType());
```

Comment:
This is intended to ensure there's no multiple binary subtype of binary _id on a single collection.
Am I reading it correctly?

Reply:
not really - this is to make sure we write the correct filter when querying mongo from record iterator